### PR TITLE
Fix #45 Make it easy to delete default channels, servers and networks

### LINKED ISSUE #45: make it easy to delete default channels, servers and networks
irssi comes with a lot of networks and servers in the default config. They are there to make it easier for new users to get started. 

Please make it easier to remove all networks and servers to get a "clean slate".

### PR DESCRIPTION
**Pull request by isundil**
Removing network will now also remove all attached servers

### CODE REVIEW

**Comment by ailin-nemui** [comment]
what about attached channels?

**Comment by dequis** [comment]
You're iterating over `server_setup_find_chatnet` with a for loop, building a new linked list with only some entries, returning that linked list, iterating over the returned linked list to remove each element, and not freeing the resulting linked list at any point.

You should iterate in a similar way to what `server_remove_channels` in src/core/servers.c does, keeping a `GSList *next` pointer and doing `tmp = next` instead of `tmp = tmp->next`

**Comment by isundil** [comment]
I added changes with your considerations.
Channels are also removed now.

**Comment by dequis** [comment]
Not sure if i like the fact that both functions look copypasted, but, different types. I dunno. Probably okay.

**Comment by isundil** [comment]
I know it look copypasted, but since the servers and channels are both linked-list their removal function should look alike.
The `channel_setup_find' and `server_setup_find' functions from these files are quite similar too.

**Comment by dequis** [comment]
This looks good, but if anyone else can have a look and merge it, would be cool.

File: src/core/servers-setup.c [hunk 1]
```diff
@@ -505,6 +505,21 @@ void server_setup_add(SERVER_SETUP_REC *rec)
 	signal_emit("server setup updated", 1, rec);
 }
 
+void server_setup_remove_chatnet(const char *chatnet)
+{
+	GSList *tmp, *next;
+
+	g_return_if_fail(chatnet != NULL);
+
+	for (tmp = setupservers; tmp != NULL; tmp = next) {
```
Comment (ailin-nemui) [inline src/core/servers-setup.c]: nit: why not the usual `tmp = tmp->next`?

Comment (dequis) [inline src/core/servers-setup.c]: See my comment above, iterating that way would result in use after free of tmp

Comment (isundil) [inline src/core/servers-setup.c]: Because I think tmp may be unallocated by `g_slist_remove` (at line 180)

Comment (ailin-nemui) [inline src/core/servers-setup.c]: I see

**Pull request merged**